openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -179,134 +179,193 @@ def check_permissions_for_scope(
             return False
-        intersection_meetings = self.datastore.get_many(
+        return self.datastore.get_many(
             [
                 GetManyRequest(
                     "meeting",
                     list(intersection_meeting_ids),
-                    ["admin_group_id", "group_ids", "meeting_user_ids"],
+                    ["admin_group_id", "group_ids"],
                 )
             ],
             lock_result=False,
         ).get("meeting", {})
-        for meeting_id, meeting_dict in intersection_meetings.items():
-            admin_meeting_users = {}
-            admin_group_id = meeting_dict.get("admin_group_id", 0)
-            admin_group = self.datastore.get(
-                fqid_from_collection_and_id("group", admin_group_id),
-                ["meeting_user_ids"],
-                lock_result=False,
-            )
-            admin_meeting_users = self.datastore.get_many(
+
+    def _collect_admin_meeting_users(
+        self, intersection_meetings: dict[int, Any]
+    ) -> set[int]:
+        """
+        Gets the admin groups and those groups with permission User.CAN_UPDATE and USER.CAN_MANAGE of the meetings.
+        Returns a set of the groups meeting_user_ids.
+        """
+        group_ids = [
+            group_id
+            for meeting_id, meeting_dict in intersection_meetings.items()
+            for group_id in meeting_dict.get("group_ids", [])
+        ]
+        admin_group_ids = [
+            meeting_dict["admin_group_id"]
+            for meeting_dict in intersection_meetings.values()
+        ]
+        return {
+            *{
+                mu_id
+                for group_id, group in self.datastore.get_many(
+                    [
+                        GetManyRequest(
+                            "group", group_ids, ["meeting_user_ids", "permissions"]
+                        )
+                    ],
+                    lock_result=False,
+                )
+                .get("group", {})
+                .items()
+                if (
+                    "user.can_update" in group.get("permissions", [])
+                    or "user.can_manage" in group.get("permissions", [])
+                )
+                for mu_id in group.get("meeting_user_ids", [])
+            },
+            *{
+                mu_id
+                for group_id, group in self.datastore.get_many(
+                    [GetManyRequest("group", admin_group_ids, ["meeting_user_ids"])],
+                    lock_result=False,
+                )
+                .get("group", {})
+                .items()
+                for mu_id in group.get("meeting_user_ids", [])
+            },
+        }
+
+    def _collect_admin_users(self, meeting_user_ids: set[int]) -> dict[int, set[int]]:
+        """
+        Helper function used in method check_for_admin_in_all_meetings.
+        Returns the corresponding users of the groups meeting_users in a defaultdict meeting_id: user_ids.
+        """
```

Comment:
Internally, this method doesn't care at all about admin status, the name is misleading.

Reply:
`collect_users` or `get_meeting_to_user_id_dict` would be a better name